### DISCUSSION TITLE
zipball/tarball confusion & need to get zipball/tarball file size information via API

### DISCUSSION THREAD

**Discussion by oyeaussie** [community/community#126140] [Apps, API and Webhooks]
### Select Topic Area

General

### Body

Ive been working with github API, trying to download latest releases files via API and been scratching my head for few hours.

When I generate a new release, it auto generates 2 files, a zip and a tar.gz file, which show up under "Assets" label on the release page. See image:
![image](https://github.com/community/community/assets/6189974/1b26ca92-bee7-4fa1-956f-ff1d5bce20a0)

When I make an API call to the latest release, the assets array is empty. As per the above image, those 2 files are under label Assets, so my assumption is that they will show up under the Assets array key when I make the API call, which does not happen.
They are zipball and tarball "archives" and to download them via API, you have to make a different call (/repos/{owner}/{repo}/zipball/{ref})

Further research revealed that assets files are something that you upload during the time of generating the release and those files will show up in the API call of latest release.

If those 2 files are "Archives" and not "Assets", why are they under the label Assets and not under label Archives?
If they are part of the Assets of a release, why don't they show up in the Assets array?

This whole thing started because I was looking for file sizes of those assets (archive) files and there is no way to get them via the API call of latest release.

I hope I am trying to explain this right.

**Comment by Syndesi** [comment]
Furthermore the archive files provided via API (`https://api.github.com/repos/owner/repo/zipball/tag`) and those linked on the release page (`https://github.com/owner/repo/archive/refs/tags/tag.zip`) are not identical, their sha256 hash does not match.
